kotest-framework/kotest-framework-multiplatform-plugin-gradle/build.gradle.kts

```diff
@@ -90,3 +73,39 @@ tasks {
       }
    }
 }
+
+
+val updateKotestPluginConstants by tasks.registering {
+   val kotestConstantsFileContent: String = """
+            |// Do not edit manually. This file was created by ${this.path}
+            |
+            |package io.kotest.framework.multiplatform.gradle
+            |
+            |const val KOTEST_COMPILER_PLUGIN_VERSION: String = "${Ci.gradleVersion}"
+            |
+         """.trimMargin()
+   inputs.property("kotestConstantsFileContent", kotestConstantsFileContent)
+
+   val kotestConstantsOutputFile = project.layout.projectDirectory.file(
```

Comment:
Rather than writing this to the main source code directory, it's probably best to write this to a file under the build directory and attach the file to the `main` source set. 

If the `kotestPluginConstants.kt` file is declared as an output file from this task, you could even attach the task's outputs to the source set, which will then ensure that any other task that requires source files will correctly depend on this task, removing the need for the `tasks.withType<KotlinCompile>().configureEach { dependsOn(updateKotestPluginConstants) }` block below.

Reply:
Good points, I'll set up a generated source set 👍

Reply:
I've set up a separate source directory. I'm not sure how to attach the task output to the source set, could you explain or show some code?

Reply:
I can't find a good example unfortunately, but I believe the way to do it is to call [`srcDir`](https://docs.gradle.org/current/javadoc/org/gradle/api/file/SourceDirectorySet.html#srcDir-java.lang.Object-) on the `main` source set, passing the `Task` (or `Provider<Task>`).

Reply:
Alright, I think I've fixed this!

This was the best example I found: https://docs.gradle.org/current/userguide/declaring_dependencies_between_subprojects.html#sec:depending_on_output_of_another_project

I also *finally* found a good use for the `resources.text.fromString(...)` Gradle util 🎉

Reply:
Resolved in https://github.com/kotest/kotest/pull/3168